components/input/gpio/gpio.go

```diff
@@ -21,6 +21,23 @@ import (
 
 const modelName = "gpio"
 
+// Config is the overall config.
+type Config struct {
+	Board   string                  `json:"board"`
+	Buttons map[string]ButtonConfig `json:"buttons"`
+	Axes    map[string]AxisConfig   `json:"axes"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (config *Config) Validate(path string) ([]string, error) {
```

Comment:
Since he's on PTO, I looked at the file, I suggest omitting them both as empty by default, and only returning an error or warning if both `Buttons` and `Axes` are empty. There can be a use case where you use the joystick or the buttons only.